### PR TITLE
DDF-1252 updated proxy lib to accept a camel bean for outgoing response and added policy rewrite bean to platform proxy

### PR DESCRIPTION
**Pull request by stustison**
@pklinef @shaundmorris @brendan-hofmann 

@AzGoalie is the hero

[Review on Reviewable](https://reviewable.io/reviews/codice/ddf-platform/63)

### CODE REVIEW

**Comment by brendan-hofmann** [comment]
I don't see any documentation updates.

**Comment by brendan-hofmann** [comment]
Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions, some commit checks broke.

---

<sup>**[platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java, line 73 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/63#-JsRZVJzZcaxlZa5PBFC)** ([raw file](https://github.com/codice/ddf-platform/blob/412ae1dd085c836b02edd9f318412033b8bbd288/platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java#L73)):</sup>
It would be better if this method threw typed exceptions instead of the generic Exception.

---

<sup>**[platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java, line 111 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/63#-JsRa0MyQk1u1hkrGINi)** ([raw file](https://github.com/codice/ddf-platform/blob/412ae1dd085c836b02edd9f318412033b8bbd288/platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java#L111)):</sup>
Don't care whether cxfConfig exists? ;)

---

<sup>**[platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java, line 115 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/63#-JsRa9-Yhkwfi4oK9LxT)** ([raw file](https://github.com/codice/ddf-platform/blob/412ae1dd085c836b02edd9f318412033b8bbd288/platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java#L115)):</sup>
Try with resources instead?

---

<sup>**[platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java, line 175 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/63#-JsRZtaX9eDyNKMRAQhN)** ([raw file](https://github.com/codice/ddf-platform/blob/412ae1dd085c836b02edd9f318412033b8bbd288/platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java#L175)):</sup>
Should any of this behavior be configurable? Not sure what the likelihood of end-users needing to rewrite some of their own policy stuff is as well.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/63)

**Comment by stustison** [comment]
Review status: 4 of 5 files reviewed at latest revision, 4 unresolved discussions, some commit checks pending.

---

<sup>**[platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java, line 73 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/63#-JsRZVJzZcaxlZa5PBFC)** ([raw file](https://github.com/codice/ddf-platform/blob/412ae1dd085c836b02edd9f318412033b8bbd288/platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java#L73)):</sup>
That is what the proxy throws.

---

<sup>**[platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java, line 175 \[r1\]](https://reviewable.io:443/reviews/codice/ddf-platform/63#-JsRZtaX9eDyNKMRAQhN)** ([raw file](https://github.com/codice/ddf-platform/blob/412ae1dd085c836b02edd9f318412033b8bbd288/platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java#L175)):</sup>
They can make their own proxy if they need someone more than what this provides.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/63)

**Comment by stustison** [comment]
It didn't change.

---







Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/63)

**Comment by brendan-hofmann** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/63)

File: platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java [hunk 1]
```diff
@@ -117,4 +130,55 @@ public void stopProxy() throws Exception {
...
+            this.httpsPort = httpsPort;
+            this.host = host;
+            this.cxfContext = cxfContext;
+        }
+
+        public void rewrite(Exchange exchange) throws IOException {
+            try {
+                Message httpMessage = exchange.getIn();
+                if (httpMessage != null) {
+                    Object body = httpMessage.getBody();
```
Comment (pklinef) [inline platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java]: It does not look like you use body outside of the check in the next line.  You can combine them.

**Comment by shaundmorris** [comment]
Reviewed 4 of 5 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks broke.

---

<sup>**[platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java, line 170 \[r2\]](https://reviewable.io:443/reviews/codice/ddf-platform/63#-JsYbdFI3TBgLnK-joLz)** ([raw file](https://github.com/codice/ddf-platform/blob/e04f86c0a6e0058b91bbb3145ed7c6df9ac33ea5/platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java#L170)):</sup>
Seems unlikely. But since this is using the "wsp" prefix any concerns that a request could come in with a different prefix?

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/63)

**Comment by stustison** [comment]
<sup>**[platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java, line 170 \[r2\]](https://reviewable.io:443/reviews/codice/ddf-platform/63#-JsYbdFI3TBgLnK-joLz)** ([raw file](https://github.com/codice/ddf-platform/blob/e04f86c0a6e0058b91bbb3145ed7c6df9ac33ea5/platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java#L170)):</sup>
No, this is modifying the response, which we control.

---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/63)

**Comment by AzGoalie** [comment]
Tried doing a full build, but it fails on `testStartingAndStoppingProxyWithDifferentConfigurations(org.codice.ddf.platform.http.proxy.TestHttpProxy)`

File: platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java [hunk 2]
```diff
@@ -81,25 +90,29 @@ public void startProxy() throws Exception {
                 host = "localhost";
             }
             endpointName = ((HttpProxyServiceImpl) httpProxyService)
-                    .start("0.0.0.0:" + httpPort, "https://" + host + ":" + httpsPort, 1000, true);
+                    .start("0.0.0.0:" + httpPort, "https://" + host + ":" + httpsPort, 120000, true,
```
Comment (shaundmorris) [inline platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java]: @stustison, @AzGoalie I think this is causing the problem. Looks like the unit tests are expecting 1000 like eq(1000), but the implementation here has 120000. So if that is still the appropriate number than I believe we need to update the unit tests.

Comment (stustison) [inline platform-http-proxy/src/main/java/org/codice/ddf/platform/http/proxy/HttpProxy.java]: Yeah that is my bad. I changed it to fix a different issue I was seeing in the integration tests and did a quick build :) I'll update the unit tests.

**Comment by stustison** [comment]
Should be fixed now.

---







Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/63)

**Comment by AzGoalie** [comment]
Reviewed 3 of 5 files at r1, 1 of 1 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-platform/63)

**Pull request merged**